File: dummy_test_generator.py
import os
import random


def create_dummy_test_files(directory, num_classes, max_test_cases_per_class):
    """
    Generate dummy test class files.
    :param directory:
    :param num_classes:
    :param max_test_cases_per_class:
    :return:
    """
    for n in range(num_classes):
        with open(os.path.join(directory, "test_class_{}.py".format(n)), "w") as f:
            content = "import unittest\n\n\nclass {}(unittest.TestCase):\n".format("Dummy" + str(n) + "Test")
            for i in range(random.randint(1, max_test_cases_per_class)):
                content += "    def test_{}(self):\n        self.assertTrue(True)\n\n".format(i)
            f.write(content)


if __name__ == '__main__':
    create_dummy_test_files("tests", 100, 20)

File: php-test-parallelization/DummyTestGenerator.php
<?php

function create_dummy_test_files($directory, $num_classes, $max_test_cases_per_class)
{
    for ($i = 0; $i < $num_classes; $i++) {
        $test_cases = "";
        for ($j = 0; $j < rand(1, $max_test_cases_per_class); $j++)
        {
            $test_cases .= <<<END
    public function test{$j}()
    {
        sleep(0);
        \$this->assertTrue(true);
    }\n\n
END;


        }

        $content = <<<END
<?php

use PHPUnit\Framework\TestCase;

class Dummy{$i}Test extends TestCase
{
$test_cases}
END;

        file_put_contents($directory . "/Dummy{$i}Test.php", $content);
    }

}

create_dummy_test_files("./tests", 100, 20);
